src/Microsoft.Extensions.Configuration.UserSecrets/UserSecretsIdAttribute.cs

```diff
@@ -19,7 +19,7 @@ public UserSecretsIdAttribute(string userSecretId)
         {
             if (string.IsNullOrEmpty(userSecretId))
             {
-                throw new ArgumentNullException(nameof(userSecretId));
+                throw new ArgumentException(nameof(userSecretId));
```

Comment:
This now needs a proper message. We often have a Common_StringNullOrEmpty message that we use everywhere. Otherwise this exception has little meaning because it doesn't say what was invalid.